### PR TITLE
WTrackMenu: move actions into Auto DJ and Track Files menus

### PR DESCRIPTION
**Pull request by ronso0**
https://mixxx.zulipchat.com/#narrow/stream/247619-UI-.26-UX-design/topic/Configurable.20Track.20Menu

### CODE REVIEW

**Comment by daschuer** [comment]
I am strongly against this because the Auto DJ top and bottom menu entries are the most used entries during a DJ session. I cannot effort the extra mouse move to a sub menu.

**Comment by daschuer** [comment]
Here is the original discussion:

https://github.com/mixxxdj/mixxx/pull/1427#issuecomment-352984139

IMHO the issue that makes it worse is the third "replace" entry. Feel free to remove this.

**Comment by ronso0** [comment]
Okay, I expected some discussion : )

> Auto DJ top and bottom menu entries are the most used entries during a DJ session

During **your** DJ session : )
IMO if we want to decide based on such arguments we should make a real survey, not amongst devs.

However we will always find someone how "really needs" this and that option, and we will get nowhere but add more and more legitimate actions to the top level of the menu.
That's why I repeatedly argued for making the menu configurable.

> What are your most frequent used menu entries?

top > bottom = most used > rarely used + (reasons I don't need other options), unmentioned actions never used
In the tracks table ():
* Open in File Browser
* Search related tracks
* Delete Track File
* Double BPM
* Hide from Library (oh I can use `Del` now : )
* (drag'n'drop for very rare playlist management, don't use crates anymore since I started 'tagging' via comments)
* (Ctrl+Return opens Properties)

In decks:
* Delete Track File
* Adjust ReplayGain from deck gain
* (open Properties via double click)

I am not motivated to repeat what I wrote on Zulip, I think it's best to discuss details there.
[Zulip: Configurable Track Menu](https://mixxx.zulipchat.com/#narrow/stream/247619-UI-.26-UX-design/topic/Configurable.20Track.20Menu)
Looking at that right now it seems we won't even reach a concensus to group the file actions :|

**Comment by ronso0** [comment]
> How about adding keyboard shortcuts for the "Add to AutoDJ" actions while moving them to a submenu?

We already have all those as `[Library],AutoDJAdd...`.

**Comment by daschuer** [comment]
A cool solution would be to add buttons to a single menu entry:
"Add to Auto DJ [↑][↓][!]" 
I just wonder if we can make this keyboard accessible.

"https://stackoverflow.com/questions/8358965/is-there-a-way-to-add-a-qwidget-to-a-qmenu-in-qtcreator

**Comment by ronso0** [comment]
> With keyboard shortcuts, we have in this case zero visual feedback, which is a usability issue.

Well, that is also the case when an AutoDJ action is selected as double-click action.
There is [`featureIsLoading`](https://github.com/mixxxdj/mixxx/blob/1879ad4f310dad8c35fbc79a59a2e4a3851ee9a0/src/library/libraryfeature.h#L136) and `featureLoadingFinished`. Could be used for a briefly highlighting the AutoDJ sidebar item.

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/widget/wtrackmenu.cpp [hunk 1]
```diff
@@ -105,6 +105,11 @@ void WTrackMenu::popup(const QPoint& pos, QAction* at) {
 }
 
 void WTrackMenu::createMenus() {
+    if (featureIsEnabled(Feature::AutoDJ)) {
+        m_pAutoDJMenu = new QMenu(this);
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/widget/wtrackmenu.cpp:109]: sure. I considered that option off the table by now anyway, the QWigetAction with 3 buttons is much better if we can make it work.
I made a prototype earlier, it works nicely with only minor changes to the AutoDJ slots. Didn't check the focus behavior and styling, yet. Should be manageable though

[the remark this answers is not in the document]
Comment (ronso0) [reply src/widget/wtrackmenu.cpp:109]: I already had a working prototype with the `[↑]` `[↓]` `[!]` buttons, even some focus/navigation issues solved ~90%.
and now I can't find it anymore... carelessly cleaned up git stash or local branches? 
meeeh.....

[the remark this answers is not in the document]
Comment (daschuer) [reply src/widget/wtrackmenu.cpp:109]: does 
```
git reflog 
```
reveal the old commits? Normally they a still there if you have at least committed them.

[the remark this answers is not in the document]
Comment (ronso0) [reply src/widget/wtrackmenu.cpp:109]: this didn't point to any usable commit, just `[sha1] HEAD@{xxxx}: checkout: moving from trackmenu-autodj-buttons to search-date-span` logs, but at least that revealed the branch name.

What helped was creating .git/lost-found.txt with
```
git fsck --full --no-reflogs --unreachable --lost-found | grep commit | cut -d\  -f3 | xargs -n 1 git log -n 1 --pretty=oneline > .git/lost-found.txt
```
in there I found the last commit, `git checkout [sha1]`
et voilà, back ontrack!

Thanks for the hint @daschuer

[the remark this answers is not in the document]
Comment (ronso0) [reply src/widget/wtrackmenu.cpp:109]: https://github.com/ronso0/mixxx/tree/auto-dj-buttons

**Pull request closed without merging**